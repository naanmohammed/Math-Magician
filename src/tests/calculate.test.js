import calculate from '../logic/calculate';

describe('calculate', () => {
  let calculatorData;

  beforeEach(() => {
    calculatorData = { total: '8', next: '2', operation: '+' };
  });

  it('returns an empty object when given "AC"', () => {
    expect(calculate(calculatorData, 'AC')).toMatchSnapshot();
  });

  it('updates next when given a number', () => {
    calculatorData = calculate(calculatorData, '3');
    expect(calculatorData).toMatchSnapshot();
  });

  it('updates next when given "0"', () => {
    calculatorData = calculate(calculatorData, '0');
    expect(calculatorData).toMatchSnapshot();
  });

  it('updates next when given "."', () => {
    calculatorData = calculate(calculatorData, '.');
    expect(calculatorData).toMatchSnapshot();
  });

  it('calculates the total when given "="', () => {
    calculatorData = calculate(calculatorData, '=');
    expect(calculatorData).toMatchSnapshot();
  });

  it('changes the sign of next or total when given "+/-"', () => {
    calculatorData = calculate(calculatorData, '+/-');
    expect(calculatorData).toMatchSnapshot();

    calculatorData = { total: '-8', next: null, operation: null };
    calculatorData = calculate(calculatorData, '+/-');
    expect(calculatorData).toMatchSnapshot();
  });

  it('updates next when given an operation button and two operands are set', () => {
    calculatorData = { total: '8', next: '2', operation: '+' };
    calculatorData = calculate(calculatorData, '-');
    expect(calculatorData).toMatchSnapshot();
  });

  it('returns an empty object when given a "." after a number with a decimal point', () => {
    calculatorData = { total: '8.5', next: null, operation: null };
    calculatorData = calculate(calculatorData, '.');
    expect(calculatorData).toMatchSnapshot();
  });
});
