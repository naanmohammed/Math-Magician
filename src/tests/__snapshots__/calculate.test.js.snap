// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculate calculates the total when given "=" 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": "10",
}
`;

exports[`calculate changes the sign of next or total when given "+/-" 1`] = `
Object {
  "next": "-2",
  "operation": "+",
  "total": "8",
}
`;

exports[`calculate changes the sign of next or total when given "+/-" 2`] = `
Object {
  "next": null,
  "operation": null,
  "total": "8",
}
`;

exports[`calculate returns an empty object when given "AC" 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": null,
}
`;

exports[`calculate returns an empty object when given a "." after a number with a decimal point 1`] = `Object {}`;

exports[`calculate updates next when given "." 1`] = `
Object {
  "next": "2.",
  "operation": "+",
  "total": "8",
}
`;

exports[`calculate updates next when given "0" 1`] = `
Object {
  "next": "20",
  "operation": "+",
  "total": "8",
}
`;

exports[`calculate updates next when given a number 1`] = `
Object {
  "next": "23",
  "operation": "+",
  "total": "8",
}
`;

exports[`calculate updates next when given an operation button and two operands are set 1`] = `
Object {
  "next": null,
  "operation": "-",
  "total": "10",
}
`;
